tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -836,17 +835,125 @@ export class WebGPUBackend extends KernelBackend {
     return output;
   }
 
+  private getFromPixelTextureLayout(useImport: boolean): WebGPULayout {
+    if (useImport) {
+      if (this.fromPixelImportTextureLayout === null) {
+        this.fromPixelImportTextureLayout =
+            this.createFromPixelTextureLayout(true);
+      }
+      return this.fromPixelImportTextureLayout;
+    }
+
+    if (this.fromPixelTextureLayout === null) {
+      this.fromPixelTextureLayout = this.createFromPixelTextureLayout(false);
+    }
+    return this.fromPixelTextureLayout;
+  }
+
+  private createFromPixelTextureLayout(useImport: boolean): WebGPULayout {
```

Comment:
Ok for this PR now. 
But overall (for all other programs) can we use Pipeline.GetBindGroupLayout() instead of such kind of function?

Reply:
This is intendted behaviour, we cache all layout, please see this change: https://github.com/tensorflow/tfjs/pull/5001/files